cmd/kubectl-gadget/utils/trace.go

```diff
@@ -531,12 +575,9 @@ func waitForCondition(traceID string, conditionFunction func(*gadgetv1alpha1.Tra
 			return false, nil
```

Comment:
This can't return `false` here all the times, otherwise if all traces are in error it implies this code would time for sure. 

I think this block can be rewritten as:

```
		if trace.Status.OperationWarning != "" {
			// The trace can have a warning but satisfies conditionFunction.
			// So, we do not add it to the map here.
			nodeWarnings[trace.Spec.Node] = trace.Status.OperationWarning
		}

		if trace.Status.OperationError != "" {
			nodeErrors[trace.Spec.Node] = trace.Status.OperationError
		}

		// If the trace does not satisfy the condition function, we are not
		// interested.
		if trace.Status.OperationError == "" && !conditionFunction(trace) {
			return false, nil
		}

		watchedTraces[trace.ObjectMeta.Name] = *trace

		return len(watchedTraces) == tracesToWatch, nil
```